### PR TITLE
Issue 43 - Fixes publishConfig.tag as well as version number on upgrade

### PR DESCRIPTION
**Pull request by alexandervalencia**
Hey @Robert-Frampton,<br><br>Attached is the fix for [Issue #43](https://github.com/liferay/liferay-theme-tasks/issues/43).<br><br>Let me know if you have any questions.<br><br>Thanks!

Also tagging @blzaugg

### CODE REVIEW

File: lib/upgrade/6.2/upgrade.js [hunk 1]
```diff
@@ -201,6 +201,14 @@ module.exports = function(options) {
 		cb();
 	});
 
+	gulp.task('upgrade:publish-config', function() {
+		var lfrThemeConfig = require('../../liferay_theme_config.js');
+
+		lfrThemeConfig.setPublishConfig({
+			tag: '7_0_x'
```
Comment (blzaugg) [inline lib/upgrade/6.2/upgrade.js]: Should it be `7_0_x` or `7_0_0`. My own issue summary confuses me.

File: lib/upgrade/6.2/upgrade.js [hunk 2]
```diff
@@ -268,6 +276,12 @@ module.exports = function(options) {
 			.pipe(gulp.dest(DIR_SRC_CSS));
 	});
 
+	gulp.task('upgrade:version', function() {
```
Comment (blzaugg) [inline lib/upgrade/6.2/upgrade.js]: I'd my more explicit. Like `upgrade:npm-package-version`.

File: lib/upgrade/6.2/upgrade.js [hunk 3]
```diff
@@ -268,6 +276,12 @@ module.exports = function(options) {
...
+		var lfrThemeConfig = require('../../liferay_theme_config.js');
+
+		lfrThemeConfig.setVersion('1.0.0');
```
Comment (blzaugg) [inline lib/upgrade/6.2/upgrade.js]: How was `0.0.0` being set before? Can we not do it there?

Should `1.0.0` even be the value, as this is the beginning of development?

File: lib/liferay_theme_config.js [hunk 4]
```diff
@@ -82,3 +82,31 @@ module.exports.setDependencies = function(dependencies, devDependencies) {
 
 	writePackageJSON(packageJSON);
 };
+
+module.exports.setPublishConfig = function(config) {
```
Comment (blzaugg) [inline lib/liferay_theme_config.js]: Should this be done/called in `setConfig`?

File: lib/liferay_theme_config.js [hunk 5]
```diff
@@ -82,3 +82,31 @@ module.exports.setDependencies = function(dependencies, devDependencies) {
...
+	}
+
+	packageJSON = _.merge(packageJSON, {
+		publishConfig: config
+	});
+
+	writePackageJSON(packageJSON);
+};
+
+module.exports.setVersion = function(versionNumber) {
```
Comment (blzaugg) [inline lib/liferay_theme_config.js]: Should this be done/called in `setConfig`?

**Comment by Robert-Frampton** [comment]
Hey guys, I think the reason we added the `publishConfig` prop was to encourage theme developers to publish their themes on npm in such a way that clarifies what the theme is compatible with, in case they ever published two versions of a theme (like a version for 6.2, and another 7.0).

The thing is, people aren't really publishing their themes to npm, so setting this property to begin with probably isn't really necessary.

File: lib/upgrade/6.2/upgrade.js (hunk 3 above)
Comment (Robert-Frampton) [reply lib/upgrade/6.2/upgrade.js]: Once a theme is created by the generator, the theme dev should have full control over the version property in the package.json. We shouldn't be modifying it.

Comment (alexandervalencia) [reply lib/upgrade/6.2/upgrade.js]: For reference: https://github.com/liferay/generator-liferay-theme/blob/master/generators/app/index.js#L316

**Pull request closed without merging**